src/cache/tlru-cache.ts

```diff
@@ -0,0 +1,56 @@
+import { LRUCache } from './lru-cache';
+
+/**
+ * Time-aware, least-recently-used (TLRU), variant of LRU where entries have valid lifetime.
+ * @param {number} maxSize - Maximum cache size
+ * @param {number} ttl - Time in milliseconds after which cache entry will evict itself
+ **/
+export class TLRUCache extends LRUCache {
+  private readonly cacheEntriesTimoutIds = new Map<string, ReturnType<typeof setTimeout>>();
+  constructor(readonly maxSize: number, readonly ttl: number) {
+    super(maxSize);
+  }
+
+  private clearCacheEntryTimeoutIfExists(key: string): void {
+    if (this.cacheEntriesTimoutIds.has(key)) {
+      const timeoutId = this.cacheEntriesTimoutIds.get(key);
+      clearTimeout(timeoutId);
+      this.cacheEntriesTimoutIds.delete(key);
+    }
+  }
+
+  private setCacheEntryTimeout(key: string): void {
+    const timeoutId = setTimeout(() => {
+      this.delete(key);
+    }, this.ttl);
+
+    this.cacheEntriesTimoutIds.set(key, timeoutId);
+  }
+
+  delete(key: string): boolean {
+    this.clearCacheEntryTimeoutIfExists(key);
+    return super.delete(key);
+  }
+
+  get(key: string): string | undefined {
+    const value = super.get(key);
+
+    if (value) {
```

Comment:
nit: due to JS falsiness rules, this would treat empty strings (`""`) as falsy and not reset their timer. Not a big issue because we currently don't use empty strings for values but would be nice to weed this small bug
```suggestion
    if (value !== undefined) {
```